contracts/contracts/vault/OETHVaultCore.sol

```diff
@@ -145,4 +156,212 @@ contract OETHVaultCore is VaultCore {
+     */
+    function requestWithdrawal(uint256 _amount)
+        external
+        whenNotCapitalPaused
+        nonReentrant
+        returns (uint256 requestId, uint256 queued)
+    {
+        // Burn the user's OETH
+        // This also checks the requester has enough OETH to burn
+        oUSD.burn(msg.sender, _amount);
+
+        requestId = withdrawalQueueMetadata.nextWithdrawalIndex;
+        queued = withdrawalQueueMetadata.queued + _amount;
+
+        // store the next withdrawal request
+        withdrawalQueueMetadata.nextWithdrawalIndex = uint128(requestId + 1);
+        withdrawalQueueMetadata.queued = uint128(queued);
+
+        emit WithdrawalRequested(msg.sender, requestId, _amount, queued);
+
+        withdrawalRequests[requestId] = WithdrawalRequest({
+            withdrawer: msg.sender,
+            claimed: false,
+            amount: uint128(_amount),
+            queued: uint128(queued)
+        });
+    }
+
+    /**
+     * @notice Claim a previously requested withdrawal once it is claimable.
+     * @param requestId Unique ID for the withdrawal request
+     * @return amount Amount of WETH transferred to the withdrawer
+     */
+    function claimWithdrawal(uint256 requestId)
+        external
+        whenNotCapitalPaused
+        nonReentrant
+        returns (uint256 amount)
+    {
+        amount = _claimWithdrawal(requestId);
+
+        // transfer WETH from the vault to the withdrawer
+        IERC20(weth).safeTransfer(msg.sender, amount);
+    }
+
+    /**
+     * @notice Claim a previously requested withdrawals once they are claimable.
+     * @param requestIds Unique ID of each withdrawal request
+     * @return amounts Amount of WETH received for each request
+     */
+    function claimWithdrawals(uint256[] memory requestIds)
+        external
+        whenNotCapitalPaused
+        nonReentrant
+        returns (uint256[] memory amounts, uint256 totalAmount)
+    {
+        amounts = new uint256[](requestIds.length);
+        for (uint256 i = 0; i < requestIds.length; ++i) {
+            amounts[i] = _claimWithdrawal(requestIds[i]);
+            totalAmount += amounts[i];
+        }
+
+        // transfer all the claimed WETH from the vault to the withdrawer
+        IERC20(weth).safeTransfer(msg.sender, totalAmount);
+    }
+
+    // slither-disable-start reentrancy-no-eth
+
+    function _claimWithdrawal(uint256 requestId)
+        internal
+        returns (uint256 amount)
+    {
+        // Check if there's enough liquidity to cover the withdrawal request
+        WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
+        WithdrawalRequest memory request = withdrawalRequests[requestId];
+
+        require(request.claimed == false, "already claimed");
+        require(request.withdrawer == msg.sender, "not requester");
+
+        // Try and get more liquidity in the withdrawal queue if there is not enough
+        if (request.queued > queue.claimable) {
```

Comment:
This branch only needs to be executed once on the first iteration of the loop line 225.